src/transformer.js

```diff
@@ -498,12 +502,15 @@ function _renderMarkdown(htmlDoc, mediaMap) {
 
             if (original !== rendered) {
                 if (mediaMap != null) {
-                    const fragment = libxmljs.parseHtmlFragment(rendered);
-
-                    rendered = _replaceMediaSources(
-                        fragment,
-                        mediaMap
-                    ).toString(false);
+                    const fragment = libxmljs.parseHtmlFragment(
+                        `<div class="temporary-root">${rendered}</div>`
+                    );
+
+                    rendered = _replaceMediaSources(fragment, mediaMap)
+                        .root()
+                        .childNodes()
+                        .map((node) => node.toString(false))
+                        .join('');
```

Comment:
It was unclear to me why this would be a fix, so I did some debugging with the test form. What I found is interesting! This appears to be working around a bug in `libxmljs.parseHtmlFragment`. I looked to find out why this would happen, it appears that internally calls `libxmljs.Document.fromHtml` and hard-codes an `implied: false` option. This solution is fine, but for future reference/context I found this also works:

```js
const doc = libxmljs.Document.fromHtml(rendered);

rendered = replaceMediaSources(doc, mediaMap)
    .get('//body')
    .childNodes()
    .join('');
```